src/commands/availability/availability.js

```diff
@@ -0,0 +1,368 @@
+        };
+
+        try {
+
+            // get the used subcommand
+            const subcommand = interaction.options.getSubcommand();
+
+            action[subcommand]();
+        } catch (error) {
+            await interaction.editReply({
+                content:   `Something went wrong. ${error}`,
+                ephemeral: false,
+            });
+        }
+    }
+};
+
+// Callbacks
+// set-unavailability
+const setUnavail = async (interaction) => {
+    const userId = interaction?.user?.id;
+    const userTag = interaction?.user?.tag;
+
+    await interaction.deferReply({ ephemeral: true });
+
+    // command should include a user
+    if (userId === undefined || userTag === undefined) {
+        await interaction.editReply({
+            ephemeral: true,
+            content:   'Could not process command'
+        });
+        return;
+    }
+
+    try {
+        const dateFrom = interaction.options.get('date-from')?.value;
+        const dateTo = interaction.options.get('date-to')?.value;
+        let timeFrom = interaction.options.get('time-from')?.value;
+        let timeTo = interaction.options.get('time-to')?.value;
+        const reason = interaction.options.get('reason')?.value;
+
+        if (dateTo && !dateFrom)
+            throw new ScheduleError('Please select a start date.');
+        if (timeTo && !timeFrom)
+            throw new ScheduleError('Please select a start time.');
+
+        // Create a start and end dayjs obj (Default times to 0:00 if empty)
+        const startUnavail = dayjs(`2024 ${dateFrom} ${timeFrom ? timeFrom : '0:00'}`);
+        const endUnavail = dayjs(`2024 ${dateTo} ${timeTo ? timeTo : '23:59'}`);
+
+        if (!dayjs(startUnavail).isValid && !dayjs(endUnavail).isValid)
+            throw new ScheduleError('Enter dates and times in proper formats');
+
+        const unavail = createUnavailability(startUnavail, endUnavail, reason);
+
+        // Print data for now
+        let reply = [
+            '```',
+            JSON.stringify(unavail, undefined, 2),
+            '```',
+        ].join('\n');
+
+        await interaction.editReply({ content: reply });
+
+        // Save data to file
+        saveUnavailability(userId, userTag, unavail, path);
+    } catch (error) {
+        if (error.name === 'ScheduleError') {
+            await interaction.editReply({
+                ephemeral: true,
+                content:   `*${error.message}*`
+            });
+        } else {
+            console.log(error);
+            await interaction.editReply({
+                ephemeral: true,
+                content:   `*Issue running command*`
+            });
+        }
+    }
+};
+
+// set-availability
+const setAvail = async (interaction) => {
+    const userId = interaction?.user?.id;
+    const userTag = interaction?.user?.tag;
+
+    await interaction.deferReply({ ephemeral: true });
+
+    // command should include a user
+    if (userId === undefined || userTag === undefined) {
```

Comment:
This check doesn't seem neccessary. If a user is required, the parameters should make it required. Otherwise, don't run the command